packages/wp-source/src/errors.ts

```diff
@@ -0,0 +1,13 @@
+export class ServerError extends Error {
+  public name = "ServerError";
+
+  public status: number;
+  public statusText: string;
+
+  constructor(message = "", status: number, statusText: string = message) {
+    super(message);
+    this.message = message;
+    this.status = status;
+    this.statusText = statusText;
+  }
+}
```

Comment:
Extending `Error` is a good idea :)

I think this can be moved to `@frontity/source` because it's a shared implementation between all the source packages.

We have to move the types of `@frontity/source` to `@frontity/source/types` (I'm sorry, this was meant to be done a long time ago) and use the `index.ts` to export things like this:

```js
import Source from "@frontity/source/types";
import { ServerError } from "@frontity/source";
```

Reply:
oh, okay, gotcha 🙂